src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -246,12 +246,41 @@ class ConstraintsChecker
         return true;
     }
 
-    template <typename T, typename U, std::enable_if_t<std::is_enum<T>::value && !std::is_pointer<U>::value, int> = 0>
+    template <typename T, typename U,
+              std::enable_if_t<std::is_enum<T>::value && !std::is_enum<U>::value && !std::is_pointer<U>::value, int> = 0>
     bool CheckConstraintMinValue(const char * itemName, T current, U expected)
     {
         return CheckConstraintMinValue(itemName, chip::to_underlying(current), expected);
     }
 
+    template <typename T, typename U,
+              std::enable_if_t<std::is_enum<T>::value && std::is_enum<U>::value && std::is_convertible<U, int>::value, int> = 0>
```

Comment:
I don't understand this part.  The only difference between this template and the next one is that this one has `std::is_convertible<U, int>::value` and the next one has `!std::is_convertible<U, int>::value`.  Gut the actual implementation of the function is the same, so why don't we just combine the two cases?

Reply:
I have removed the `std::is_convertible<U, int>::value` to differenciate between weak and class enums.